lib/src/feature/add_baby/view/add_baby_screen.dart

```diff
@@ -129,61 +156,82 @@ class AddBabyScreen extends StatelessWidget {
   }
 
   Widget _renderBottomButton(BuildContext context) {
-    return XBottomButtons(
-        positiveButtonText: S.of(context).save,
-        cancelButtonText: S.of(context).cancel,
-        onTappedPositive: () {
-          //TODO: Ontap Save
-        },
-        onTappedCancel: () {
-          AppCoordinator.pop();
-        });
+    return BlocSelector<AddBabyBloc, AddBabyState, AddBabyScreenStatus>(
+      selector: (state) => state.status,
+      builder: (context, status) {
+        return XBottomButtons(
+            positiveButtonText: S.of(context).save,
+            cancelButtonText: S.of(context).cancel,
+            isLoading: status == AddBabyScreenStatus.saving,
+            onTappedPositive: () {
+              context.read<AddBabyBloc>().saveBabyInfor(context);
+            },
+            onTappedCancel: () {
+              AppCoordinator.pop();
+            });
+      },
+    );
   }
 
   Widget _renderBirthDate(BuildContext context) {
     return Expanded(
-        child: XLabelButton(
-      onTapped: () async {
-        await showBoardDateTimePicker(
-          context: context,
-          pickerType: DateTimePickerType.date,
-          options: BoardDateTimeOptions(
-            boardTitle: S.of(context).selectDate,
-            activeColor: AppColors.primary,
-          ),
+        child: BlocBuilder<AddBabyBloc, AddBabyState>(
+      buildWhen: (previous, current) =>
+          previous.babyBirthDate != current.babyBirthDate ||
+          previous.errorBabyBirthDate != current.errorBabyBirthDate,
+      builder: (context, state) {
+        return XLabelButton(
+          onTapped: () async {
+            context.read<AddBabyBloc>().showBottomSheetDateTimePicker(context,
+                title: S.of(context).selectDate,
+                type: DateTimePickerType.date,
+                onChanged: (babyBirthDate) => context
+                    .read<AddBabyBloc>()
+                    .onChangedBabyBirthDate(babyBirthDate));
+          },
+          hint: S.of(context).selectDate,
+          label: S.of(context).dateOfBirth,
+          value: isNullOrEmpty(state.babyBirthDate)
+              ? null
+              : DateFormat("MMM d, y").format(state.babyBirthDate!),
```

Comment:
Should separate into other file with meaning format date. Example: `DateTimeFormat.formatABC()`